src/argilla/server/search_engine/commons.py

```diff
@@ -211,7 +152,38 @@ def es_mapping_for_metadata_property(metadata_property: MetadataProperty) -> dic
     elif property_type == MetadataPropertyType.float:
         return {es_field_for_metadata_property(metadata_property): {"type": "float"}}
     else:
-        raise ValueError(f"Index configuration for metadata property of type {property_type} cannot be generated")
+        raise Exception(f"Index configuration for metadata property of type {property_type} cannot be generated")
+
+
+def es_mapping_for_question_type(question_type: QuestionType):
```

Comment:
Maybe I'm missing something (tell me if I'm wrong) but should not be this function better named `es_mapping_for_question` and receive a question instead of the type?

```suggestion
def es_mapping_for_question(question: Question):
```

Inside you can do the `if question.type` conditional.